src/Symfony/Component/DependencyInjection/Loader/Configurator/AbstractServiceConfigurator.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+
+abstract class AbstractServiceConfigurator extends AbstractConfigurator
+{
+    protected $parent;
+    protected $definition;
+    protected $id;
+    protected $defaultTags = array();
+
+    public function __construct(AbstractConfigurator $parent, Definition $definition, $id = null, array $defaultTags = array())
+    {
+        $this->parent = $parent;
+        $this->definition = $definition;
+        $this->id = $id;
+        $this->defaultTags = $defaultTags;
+    }
+
+    public function __destruct()
+    {
+        // default tags should be added last
+        foreach ($this->defaultTags as $name => $attributes) {
+            foreach ($attributes as $attributes) {
+                $this->definition->addTag($name, $attributes);
+            }
+        }
+        $this->defaultTags = array();
+    }
+
+    /**
+     * Registers a service.
+     *
+     * @param string      $id
+     * @param string|null $class
+     *
+     * @return ServiceConfigurator
+     */
+    final public function set($id, $class = null)
+    {
+        $this->__destruct();
+
+        return $this->parent->set($id, $class);
+    }
+
+    /**
+     * Creates an alias.
+     *
+     * @param string $id
+     * @param string $ref
+     *
+     * @return AliasConfigurator
+     */
+    final public function alias($id, $referencedId)
+    {
+        $this->__destruct();
+
+        return $this->parent->alias($id, $referencedId);
+    }
+
+    /**
+     * Registers a PSR-4 namespace using a glob pattern.
+     *
+     * @param string $namespace
+     * @param string $resource
+     *
+     * @return PrototypeConfigurator
+     */
+    final public function load($namespace, $resource)
+    {
+        $this->__destruct();
+
+        return $this->parent->load($namespace, $resource);
+    }
+
+    /**
+     * Gets an already defined service definition.
+     *
+     * @param string $id
+     *
+     * @return ServiceConfigurator
+     *
+     * @throws ServiceNotFoundException if the service definition does not exist
+     */
+    final public function get($id)
+    {
+        $this->__destruct();
+
+        return $this->parent->get($id);
+    }
+
+    /**
+     * Registers a service.
+     *
+     * @param string      $id
+     * @param string|null $class
+     *
+     * @return ServiceConfigurator
+     */
+    final public function __invoke($id, $class = null)
+    {
+        $this->__destruct();
+
+        return $this->parent->set($id, $class);
```

Comment:
$this-parent is defined as `AbstractConfigurator` but this class does not have a `set` method. So it does not seem safe to call this.

Reply:
good catch, parent is now a `ServicesConfigurator` so the issue is gone